components/script/dom/htmlformelement.rs

```diff
@@ -253,6 +255,92 @@ impl HTMLFormElementMethods for HTMLFormElement {
         let elements = self.Elements();
         elements.IndexedGetter(index)
     }
+
+    // https://html.spec.whatwg.org/multipage/#the-form-element%3Adetermine-the-value-of-a-named-property
+    fn NamedGetter(&self, name: DOMString) -> Option<RadioNodeListOrElement> {
+        // return Option::Some::<RadioNodeListOrElement>;
+        unimplemented!();
+    }
+
+    // https://html.spec.whatwg.org/multipage/#the-form-element:supported-property-names
+    fn SupportedPropertyNames(&self) -> Vec<DOMString> {
+        enum SourcedNameSource {
+            Id,
+            Name,
+            Past(std::time::Duration),
+        }
+
+        struct SourcedName {
+            name: DOMString,
+            element: DomRoot<Element>,
+            source: SourcedNameSource,
+        }
+
+        // vector to store sourced names tuple information
+        let mut sourcedNamesVec: Vec<SourcedName> = Vec::new();
+
+        // let controls = self.controls.borrow_mut(); - line 849 → unsure of which "borrow" to use
+        let controls = self.controls.borrow(); // line 807 in this file
+
+        // controls - list of form elements
+        // check all listed elements first, push to sourcedNamesVec as per spec
+        for child in controls.iter() {
+            if child.is_html_element() {
```

Comment:
@jdm Is the function call `is_html_element()` correct or we should use `is_listed()`? We weren't able to call the method `is_listed_element()` as mentioned by you on the project page. `./mach check` was showing error for function call on incorrect variable type.

Reply:
Ah, that's HTMLElement::is_listed_element. You can do:
```
if child.downcast::<HTMLElement>().map_or(false, |c| c.is_listed_element()) {
```

Reply:
Done.